mycodo/controllers/controller_pid.py

```diff
@@ -327,36 +361,29 @@ def setup_method(self, method_id):
         if self.method_type == 'Duration':
             if self.method_start_act == 'Ended':
                 # Method has ended and hasn't been instructed to begin again
-                pass
+                with session_scope(MYCODO_DB_PATH) as db_session:
+                    mod_pid = db_session.query(PID)
+                    mod_pid = mod_pid.filter(
+                        PID.unique_id == self.unique_id).first()
+                    mod_pid.is_activated = False
+                    db_session.commit()
```

Comment:
Previous implementation did database access to controller tables partly in the controller and partly in the method. The part that was done in method has been migrated to the controllers.